src/asset.ts

```diff
@@ -0,0 +1,15 @@
+export enum Asset {
+  Bitcoin = "bitcoin",
+  Ether = "ether"
+}
+
+export function toAsset(asset: string) {
```

Comment:
I'll look into it but tsc does not like your snippet of code:

![image](https://user-images.githubusercontent.com/300805/60408931-63d1b200-9c04-11e9-8d38-cb77fdde4a6b.png)

Reply:
Your mistake lies in the Asset enum. 
```
    enum Asset{
      bitcoin,
      ethereum
    }
```

vs

```
    enum Asset{
      Bitcoin = "bitcoin",
      Ethereum = "ethereum"
    }
```

Reply:
Unfortunately this cannot be achieved with `keyof typeof Asset` because it would require a typecast using `<...>`, which tsc does not like. One can indeed achieve this behaviour (without linter & Co complaints...) by doing this: 
```
const alphaAsset: Asset = (Asset as any)[swap.properties.parameters.alpha_asset.name];
const betaAsset: Asset = (Asset as any)[swap.properties.parameters.beta_asset.name];
```
However, I find this confusing and not readable. One does not understand what is going on there without looking this up, which is timely. 

Thus, I vote for leaving the explicit conversion function in the code. It is way easier to understand that way. If we overlooked another possibility to achieve this please let me know :) 

Details: https://stackoverflow.com/questions/17380845/how-do-i-convert-a-string-to-enum-in-typescript

Reply:
Sure, it was just a recommendation. Did not want to escalate this :D

Reply:
The issue is that the `(Asset as any)` actually did not work well either :)